network/server.go

```diff
@@ -673,3 +723,14 @@ func (s *Server) CountConnections() int {
 	defer s.mu.RUnlock()
 	return int(s.connections)
 }
+
+// GetProxyForConnection returns the proxy associated with the given connection.
+func (s *Server) GetProxyForConnection(conn *ConnWrapper) (IProxy, bool) {
+	proxy, exists := s.connectionToProxyMap[conn]
+	return proxy, exists
+}
```

Comment:
Nothing in your PR uses the exist value returned


```suggestion
func (s *Server) GetProxyForConnection(conn *ConnWrapper) IProxy {
	proxy, _ := s.connectionToProxyMap[conn]
	return proxy
}
```


@sinadarbouy

Reply:
added for each reference in the lines above in `server.go` to handle logging and errors if the proxy is not found